src/key.rs

```diff
@@ -282,6 +282,15 @@ impl RSAPrivateKey {
         k
     }
 
+    /// Extract the public key from the private key, cloning `n` and `e`.
+    ///
+    /// Generally this is not needed since `RSAPrivateKey` implements the `PublicKey` trait,
+    /// but it can occationally be useful to discard the private information entirely.
+    pub fn extract_public(&self) -> RSAPublicKey {
```

Comment:
I would prefer `to_public_key` to keep with rust naming conventions

Reply:
Done!